(Memory Leak?) Macro RESET_RUNTIME uses memset() on non-trivial type

This compiler warning appears legitimate.  What is the macro really attempting to do here?

<details><summary>Compiler puts out the following warning</summary><P/>

```
In file included from wled00\fcn_declare.h:132,
                 from wled00\wled.h:145,
                 from wled00\FX_fcn.cpp:26:
wled00\FX_fcn.cpp: In member function 'void WS2812FX::finalizeInit()':
wled00\FX.h:84:80: warning: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'WS2812FX::segment_runtime' {aka 'class WS2812FX::Segment_runtime'}; use assignment or value-initialization instead [-Wclass-memaccess]
   84 | #define RESET_RUNTIME    memset(_segment_runtimes, 0, sizeof(_segment_runtimes))
      |                                                                                ^
wled00\FX_fcn.cpp:70:3: note: in expansion of macro 'RESET_RUNTIME'
Retrieved `.pio\build\esp8266_2m\libb18\IRremoteESP8266\ir_Lutron.cpp.o' from cache
   70 |   RESET_RUNTIME;
      |   ^~~~~~~~~~~~~
In file included from wled00\fcn_declare.h:132,
                 from wled00\wled.h:145,
                 from wled00\FX_fcn.cpp:26:
wled00\FX.h:362:20: note: 'WS2812FX::segment_runtime' {aka 'class WS2812FX::Segment_runtime'} declared here
  362 |     typedef struct Segment_runtime { // 28 bytes
      |                    ^~~~~~~~~~~~~~~
```

</details>

The compiler appears to be correctly warning about a bad practice.

Put simply, the `Segment_runtime` structure contains an embedded pointer ... namely member `byte * data`.  Therefore, it should never be blindly `memset` to zero.

It appears that this will cause a memory leak, if there is data currently allocated (as evidenced by a non-`nullptr` value in member `data`).


Note: I got this error building default boards using platform io.
(I believe I saw this error on esp8266_2m)

This macro is only used in one location: FX_fnc.cpp at `WS2812FX::finalizeInit(void)`.   I'm not entirely sure what the "correct" fix is.  Perhaps this was just missed when fixing earlier multi-core issues with resetting of `_segment_runtimes`?  That said, here's a strawman to start discussion...

Remove the macro definition, and replace it's one use with:
```c++
  for (uint16_t i = 1; i < MAX_NUM_SEGMENTS; i++)
  {
    _segment_runtimes[i].reset();
  }
  _segment_runtimes[0].reset();
```

It's also odd ... two places that reset this loop from 1..N, then reset index zero, while a third just loops through them all.  This suggests that there may be latent bugs, such as by relying on an order of writes that's not actually guaranteed by the compiler?

--- a/wled00/FX.h
+++ b/wled00/FX.h
@@ -81,7 +81,6 @@
 #define SEGLEN           _virtualSegmentLength
 #define SEGACT           SEGMENT.stop
 #define SPEED_FORMULA_L  5U + (50U*(255U - SEGMENT.speed))/SEGLEN
-#define RESET_RUNTIME    memset(_segment_runtimes, 0, sizeof(_segment_runtimes))
 
 // some common colors
 #define RED        (uint32_t)0xFF0000
@@ -409,8 +408,9 @@ class WS2812FX {
        * Flags that before the next effect is calculated,
        * the internal segment state should be reset. 
        * Call resetIfRequired before calling the next effect function.
+       * Safe to call from interrupts and network requests.
        */
-      inline void reset() { _requiresReset = true; }
+      inline void markForReset() { _requiresReset = true; }
       private:
         uint16_t _dataLen = 0;
         bool _requiresReset = false;

--- a/wled00/FX_fcn.cpp
+++ b/wled00/FX_fcn.cpp
@@ -67,7 +67,12 @@
 //do not call this method from system context (network callback)
 void WS2812FX::finalizeInit(void)
 {
-  RESET_RUNTIME;
+  //reset segment runtimes
+  for (uint8_t i = 0; i < MAX_NUM_SEGMENTS; i++) {
+    _segment_runtimes[i].markForReset();
+    _segment_runtimes[i].resetIfRequired();
+  }
+
   _hasWhiteChannel = _isOffRefreshRequired = false;
 
   //if busses failed to load, add default (fresh install, FS issue, ...)
@@ -373,7 +378,7 @@ void WS2812FX::setMode(uint8_t segid, uint8_t m) {
 
   if (_segments[segid].mode != m) 
   {
-    _segment_runtimes[segid].reset();
+    _segment_runtimes[segid].markForReset();
     _segments[segid].mode = m;
   }
 }
@@ -612,12 +617,12 @@ void WS2812FX::setSegment(uint8_t n, uint16_t i1, uint16_t i2, uint8_t grouping,
     seg.spacing = spacing;
   }
 	if (offset < UINT16_MAX) seg.offset = offset;
-  _segment_runtimes[n].reset();
+  _segment_runtimes[n].markForReset();
 }
 
 void WS2812FX::restartRuntime() {
   for (uint8_t i = 0; i < MAX_NUM_SEGMENTS; i++) {
-    _segment_runtimes[i].reset();
+    _segment_runtimes[i].markForReset();
   }
 }
 
@@ -648,9 +653,9 @@ void WS2812FX::resetSegments() {
     _segments[i].cct = 127;
     _segments[i].speed = DEFAULT_SPEED;
     _segments[i].intensity = DEFAULT_INTENSITY;
-    _segment_runtimes[i].reset();
+    _segment_runtimes[i].markForReset();
   }
-  _segment_runtimes[0].reset();
+  _segment_runtimes[0].markForReset();
 }
 
 void WS2812FX::makeAutoSegments() {

